Using the [metapub](https://pypi.org/project/metapub/) package to fetch PubMed data.

Install using `pip`.

In [1]:
import sys
!{sys.executable} -m pip install metapub

Set NCBI API Key, which is optional but allows you to perform more queries per second (10 instead of 3).

In [2]:
!export NCBI_API_KEY="your_key"

Fetch PubMed IDs (default is up to 250).

In [3]:
from metapub import PubMedFetcher
fetch = PubMedFetcher()

keyword="checkpoint inhibitor"

pmids = fetch.pmids_for_query(keyword)
print(len(pmids))

2023-05-19 05:01:12 d315a1023584 metapub.config[1404] WARNING NCBI_API_KEY was not set.


250


Fetch PubMed IDs by specifying `retmax`.

In [4]:
from metapub import PubMedFetcher
fetch = PubMedFetcher()

keyword="checkpoint inhibitor"

pmids = fetch.pmids_for_query(keyword, retmax=100000)
print(len(pmids))

9999


9999 IDs is the maximum number that will be returned.

In [5]:
from metapub import PubMedFetcher
fetch = PubMedFetcher()

keyword="bioinformatics"

pmids = fetch.pmids_for_query(keyword, retmax=100000)
print(len(pmids))

9999


Get articles. Note that without a NCBI API KEY, your queries will be automatically throttled to 3 queries per second.

In [6]:
articles = {}
for pmid in pmids[0:5]:
    print(pmid)
    articles[pmid] = fetch.article_by_pmid(pmid)
    
print(articles[pmids[0]])

37202831
37202802
37202796
37202791
37202739
<PubMedArticle 37202831> Paskett ED; Kruse-Diehr AJ; Oliveri JM; Vanderpool RC; Gray DM; Pennell ML; Huang B; Young GS; Fickle D; Cromo M; Katz ML; Reiter PL; Rogers M; Gross DA; Fairchild V; Xu W; Carman A; Walunis JM; McAlearney AS; Huerta TR; Rahurkar S; Biederman E; Dignan M. Accelerating Colorectal Cancer Screening and Follow-up through Implementation Science (ACCSIS) in Appalachia: protocol for a group randomized, delayed intervention trial.. Transl Behav Med. 2023. None:None


Print attributes.

In [7]:
article = articles[pmids[0]]
print(f"Title: {article.title}")
print(f"Abstract: {article.abstract}")
print(f"Author/s: {article.authors}")
print(f"Published: {article.journal}, {article.year}, {article.volume}, {article.issue}")
print(f"Citation: {article.citation}")

Title: Accelerating Colorectal Cancer Screening and Follow-up through Implementation Science (ACCSIS) in Appalachia: protocol for a group randomized, delayed intervention trial.
Abstract: Appalachian regions of Kentucky and Ohio are hotspots for colorectal cancer (CRC) mortality in the USA. Screening reduces CRC incidence and mortality; however, screening uptake is needed, especially in these underserved geographic areas. Implementation science offers strategies to address this challenge. The aim of the current study was to conduct multi-site, transdisciplinary research to evaluate and improve CRC screening processes using implementation science strategies. The study consists of two phases (Planning and Implementation). In the Planning Phase, a multilevel assessment of 12 health centers (HC) (one HC from each of the 12 Appalachian counties) was conducted by interviewing key informants, creating community profiles, identifying HC and community champions, and performing HC data inventori

Build link.

In [8]:
def pubmed_url(pmid):
    return("https://pubmed.ncbi.nlm.nih.gov/" + pmid)

pubmed_url(pmids[0])

'https://pubmed.ncbi.nlm.nih.gov/37202831'